code/game/machinery/porta_turret/portable_turret.dm

```diff
@@ -177,68 +192,79 @@
 	remove_control()
 	return ..()
 
-/obj/machinery/porta_turret/ui_interact(mob/user)
-	. = ..()
+/obj/machinery/porta_turret/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, \
+									datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "PortableTurret", name, ui_x, ui_y, master_ui, state)
+		ui.open()
 
-	var/list/dat = list("Status: <a href='?src=[REF(src)];power=1'>[on ? "On" : "Off"]</a><br>")
-	dat += "Behaviour controls are [locked ? "locked" : "unlocked"]<br>"
-
-	if(!locked)
-		dat += "Check for Weapon Authorization: <A href='?src=[REF(src)];operation=authweapon'>[turret_flags & TURRET_FLAG_AUTH_WEAPONS ? "Yes" : "No"]</A><BR>"
-		dat += "Neutralize Wanted Criminals: <A href='?src=[REF(src)];operation=shootcriminals'>[turret_flags & TURRET_FLAG_SHOOT_CRIMINALS ? "Yes" : "No"]</A><BR>"
-		dat += "Neutralize All Non-Security and Non-Command Personnel: <A href='?src=[REF(src)];operation=shootall'>[turret_flags & TURRET_FLAG_SHOOT_ALL ? "Yes" : "No"]</A><BR>"
-		dat += "Neutralize All Unidentified Life Signs: <A href='?src=[REF(src)];operation=checkxenos'>[turret_flags & TURRET_FLAG_SHOOT_ANOMALOUS ? "Yes" : "No"]</A><BR>"
-		dat += "Neutralize All Non-Mindshielded Personnel: <A href='?src=[REF(src)];operation=checkloyal'>[turret_flags & TURRET_FLAG_SHOOT_UNSHIELDED ? "Yes" : "No"]</A><BR>"
-		dat += "Neutralize All Cyborgs: <A href='?src=[REF(src)];operation=shootborgs'>[turret_flags & TURRET_FLAG_SHOOT_BORGS ? "Yes" : "No"]</A><BR>"
-		dat += "Ignore Heads Of Staff: <A href='?src=[REF(src)];operation=shootheads'>[turret_flags & TURRET_FLAG_SHOOT_HEADS ? "No" : "Yes"]</A><BR>"
+/obj/machinery/porta_turret/ui_data(mob/user)
+	var/list/data = list()

```

Comment:
if you cared to you could inline the assignment of all those variables inside the list, aside from the complex assignment for the silicon stuff

ie
```dm
var/list/data = (
    "locked" = locked,
    "on" = on,
    ...
)

Reply:
seems pointless, matter of taste

Reply:
if you can do it, do it imo.

Reply:
Done.